tests/components/plex/test_server.py

```diff
@@ -100,72 +99,110 @@
+            await hass.async_block_till_done()
 
-    sensor = hass.states.get("sensor.plex_plex_server_1")
-    assert sensor.state == "0"
+        server_id = mock_plex_server.machineIdentifier
 
-
-async def test_debouncer(hass, caplog):
-    """Test debouncer decorator logic."""
-    entry = MockConfigEntry(
-        domain=DOMAIN,
-        data=DEFAULT_DATA,
-        options=DEFAULT_OPTIONS,
-        unique_id=DEFAULT_DATA["server_id"],
-    )
-
-    mock_plex_server = MockPlexServer(config_entry=entry)
-
-    with patch("plexapi.server.PlexServer", return_value=mock_plex_server), patch(
-        "homeassistant.components.plex.PlexWebsocket.listen"
-    ):
-        entry.add_to_hass(hass)
-        assert await hass.config_entries.async_setup(entry.entry_id)
+        async_dispatcher_send(hass, PLEX_UPDATE_PLATFORMS_SIGNAL.format(server_id))
         await hass.async_block_till_done()
 
-    server_id = mock_plex_server.machineIdentifier
+        sensor = hass.states.get("sensor.plex_plex_server_1")
+        assert sensor.state == str(len(mock_plex_server.accounts))
 
-    # First two updates are skipped
-    async_dispatcher_send(hass, PLEX_UPDATE_PLATFORMS_SIGNAL.format(server_id))
-    await hass.async_block_till_done()
-    async_dispatcher_send(hass, PLEX_UPDATE_PLATFORMS_SIGNAL.format(server_id))
-    await hass.async_block_till_done()
-    async_dispatcher_send(hass, PLEX_UPDATE_PLATFORMS_SIGNAL.format(server_id))
-    await hass.async_block_till_done()
+        mock_plex_server.clear_clients()
+        mock_plex_server.clear_sessions()
 
-    next_update = dt_util.utcnow() + timedelta(seconds=DEBOUNCE_TIMEOUT)
-    async_fire_time_changed(hass, next_update)
-    await hass.async_block_till_done()
+        await self.advance(DEBOUNCE_TIMEOUT)
+        async_dispatcher_send(hass, PLEX_UPDATE_PLATFORMS_SIGNAL.format(server_id))
+        await hass.async_block_till_done()
 
-    assert (
-        caplog.text.count(f"Throttling update of {mock_plex_server.friendlyName}") == 2
-    )
+        sensor = hass.states.get("sensor.plex_plex_server_1")
+        assert sensor.state == "0"
+
+    async def test_debouncer(self):
+        """Test debouncer behavior."""
+        hass = self.hass
+
+        entry = MockConfigEntry(
+            domain=DOMAIN,
+            data=DEFAULT_DATA,
+            options=DEFAULT_OPTIONS,
+            unique_id=DEFAULT_DATA["server_id"],
+        )
+
+        mock_plex_server = MockPlexServer(config_entry=entry)
+
+        with patch("plexapi.server.PlexServer", return_value=mock_plex_server), patch(
+            "homeassistant.components.plex.PlexWebsocket.listen"
+        ):
+            entry.add_to_hass(hass)
+            assert await hass.config_entries.async_setup(entry.entry_id)
+            await hass.async_block_till_done()
+
+        server_id = mock_plex_server.machineIdentifier
+
+        with patch(
+            "homeassistant.components.plex.server.PlexServer._fetch_platform_data",
```

Comment:
We should patch the api calls instead of our code here.

https://github.com/home-assistant/core/blob/5ff50e8b4f4d97c06653dd2152f02aaa3a34441e/homeassistant/components/plex/server.py#L193

Reply:
This test is only for validating the debouncer is working as expected. I patched a level higher just so I could get a call count for the debounced method.

Reply:
We should not patch our own server code in the tests for the server module.